axis.go

```diff
@@ -341,146 +351,103 @@ func (a *verticalAxis) GlyphBoxes(*Plot) (boxes []GlyphBox) {
 		}
 		boxes = append(boxes, box)
 	}
-	return
+	return boxes
 }
 
 // DefaultTicks is suitable for the Tick.Marker field of an Axis,
-// it returns a resonable default set of tick marks.
+// it returns a reasonable default set of tick marks.
 type DefaultTicks struct{}
 
 var _ Ticker = DefaultTicks{}
 
-// Ticks returns Ticks in a specified range
-func (DefaultTicks) Ticks(min, max float64) (ticks []Tick) {
-	const suggestedTicks = 3
-
-	var labels []float64
-
+// Ticks returns Ticks in the specified range.
+func (DefaultTicks) Ticks(min, max float64) []Tick {
 	if max <= min {
 		panic("illegal range")
 	}
 
-	tens := math.Pow10(int(math.Floor(math.Log10(max - min))))
-	n := (max - min) / tens
-	for n < suggestedTicks {
-		tens /= 10
-		n = (max - min) / tens
+	const suggestedTicks = 3
+
+	labels, step, q, mag := talbotLinHanrahan(min, max, suggestedTicks, withinData, nil, nil, nil)
+	majorDelta := step * math.Pow10(mag)
+	if q == 0 {
+		// Simple fall back was chosen, so
+		// majorDelta is the label distance.
+		majorDelta = labels[1] - labels[0]
 	}
 
-	majorMult := int(n / suggestedTicks)
-	switch majorMult {
-	case 7:
-		majorMult = 6
-	case 9:
-		majorMult = 8
+	// Choose a reasonable, but ad
+	// hoc formatting for labels.
```

Comment:
this 2-line comment could perhaps be on just 1 line.

Reply:
Style.